.github/workflows/build_and_test.yml

```diff
@@ -366,7 +366,9 @@ jobs:
         #   See also https://github.com/sphinx-doc/sphinx/issues/7551.
         # Jinja2 3.0.0+ causes error when building with Sphinx.
         #   See also https://issues.apache.org/jira/browse/SPARK-35375.
-        python3.6 -m pip install flake8 pydata_sphinx_theme mypy numpydoc 'jinja2<3.0.0' 'black==21.5b2'
+        # TODO(SPARK-35684, SPARK-35683): Bump up the mypy version. This is blocked by
+        #   Index.difference logic issue.
+        python3.6 -m pip install flake8 pydata_sphinx_theme 'mypy==0.812' numpydoc 'jinja2<3.0.0' 'black==21.5b2'
```

Comment:
This is only for Python3.6, right? Otherwise, we need to pin `mypy` in `dev/requirements.txt`, too.

Reply:
Actually..  the errors are different from different python versions ... e.g.) the same version with Python 3.8 doesn't pass the mypy test .. we should do something here ..